From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nickolay Savchenko <n.savchenko@axioma.lv>
Date: Thu, 16 Jul 2026 03:35:29 +0300
Subject: [PATCH] dt-bindings: net: wireless: document rt2x00 board properties

Document the board transmit-chain count and direct LED controls used by
RT3091 PCI functions with RT5392 radios.

Signed-off-by: Nickolay Savchenko <n.savchenko@axioma.lv>
---
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/wireless/ralink,rt2x00-pci.yaml
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/net/wireless/ralink,rt2x00-pci.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Ralink rt2x00 PCI wireless board properties
+
+maintainers:
+  - Nickolay Savchenko <n.savchenko@axioma.lv>
+
+description: |
+  Board-specific properties for the Ralink RT3091 PCI function with an RT5392
+  radio handled by the rt2x00 driver.
+
+select:
+  properties:
+    compatible:
+      contains:
+        const: pci1814,3091
+  required:
+    - compatible
+
+allOf:
+  - $ref: ieee80211.yaml#
+
+properties:
+  compatible:
+    const: pci1814,3091
+
+  reg:
+    maxItems: 1
+
+  ralink,mtd-eeprom:
+    $ref: /schemas/types.yaml#/definitions/phandle-array
+    items:
+      - items:
+          - description: MTD partition containing calibration data
+          - description: Offset of the calibration data in the partition
+    description: Location of the factory EEPROM data.
+
+  ralink,tx-chains:
+    $ref: /schemas/types.yaml#/definitions/uint32
+    minimum: 1
+    maximum: 2
+    description: |
+      Number of transmit chains connected by the board RF design. Receive
+      chains are described independently by the calibration data.
+
+  ralink,direct-led-control:
+    type: boolean
+    description: |
+      Program the RT5392 GPIO and LED registers directly instead of using MCU
+      LED commands. This is required by boards whose radio LED is connected to
+      the direct GPIO1 and LED_CFG path.
+
+required:
+  - compatible
+  - reg
+
+additionalProperties: false
+
+...
